File: exapi.py
import requests
import random

class ExecrossAPI:
    def __init__(self, base_url, apikey):
        self.base_url = base_url
        self.apikey = apikey
        self.session = requests.Session()
        self.session.headers.update({'apikey': self.apikey, 'Content-Type': 'application/json'})

    def getProxies(self):
        endpoint = f'{self.base_url}/proxies'
        response = self.session.get(endpoint).json()
        if response and response['status'] == 200:
            proxies = response['result']['proxies']
            return random.choice(proxies).split(':')[0]
        return None

    def getQR(self, params):
        url = f'{self.base_url}/loginqr'
        url2 = f'{self.base_url}/reqpin'
        url3 = f'{self.base_url}/reqtoken'
        response = self.session.get(url, params=params).json()
        print(response)
        if response and response['status'] == 200:
            params2 = {
                'apikey': self.apikey,
                'session': response['result']['session']
            }
            response2 = self.session.get(url2, params=params2).json()
            if response2 and 'authToken' in response2['result']:
                return response2
            else:
                print(response2)
                response3 = self.session.get(url3, params=params2).json()
                return response3
        else:
            print(f"Error: {response.get('error', 'Unknown error')}")

    def getQRV1(self, params):
        url = f'{self.base_url}/loginqrv1'
        url2 = f'{self.base_url}/reqpinv1'
        url3 = f'{self.base_url}/reqtokenv1'
        response = self.session.get(url, params=params).json()
        print(response)
        if response['status'] == 200:
            params2 = {
                'apikey': self.apikey,
                'session': response['result']['session']
            }
            response2 = self.session.get(url2, params=params2).json()
            if 'authToken' in response2['result']:
                return response2
            else:
                print(response2)
                response3 = self.session.get(url3, params=params2).json()
                return response3
        else:
            print(f"Error: {response.get('error', 'Unknown error')}")

    def getEmail(self, params):
        url = f'{self.base_url}/loginemail'
        url2 = f'{self.base_url}/reqemailtoken'
        response = self.session.get(url, params=params).json()
        print(response)
        if "pincode" in response['result']:
            params2 = {
                'apikey': self.apikey,
            }
            response2 = self.session.get(url2, params=params2).json()
            return response2

    def getEmailV1(self, params):
        url = f'{self.base_url}/loginemailv1'
        url2 = f'{self.base_url}/reqemailtokenv1'
        response = self.session.get(url, params=params).json()
        print(response)
        if "pincode" in response['result']:
            params2 = {
                'apikey': self.apikey,
            }
            response2 = self.session.get(url2, params=params2).json()
            return response2

    def exampleAddFriendPrimary(self, params):
        url = f'{self.base_url}/addfriend'
        response = self.session.get(url, params=params).json()
        return response

    def exampleAddFriendSecondary(self, params):
        url = f'{self.base_url}/addfriend'
        response = self.session.get(url, params=params).json()
        return response

    def refreshToken(self, params):
        url = f'{self.base_url}/refreshtoken'
        response = self.session.get(url, params=params).json()
        return response

    def lineAppname(self):
        url = f'{self.base_url}/appnames'
        response = self.session.get(url).json()
        if response['status'] == 200:
           return response['result']['appNames']

    def getStory(self, params):
        url = f'{self.base_url}/linestory'
        response = self.session.get(url, params=params).json()
        return response

    def getPost(self, params):
        url = f'{self.base_url}/linepost'
        response = self.session.get(url, params=params).json()
        return response 

    def friendRecomendation(self, params):
        url = f'{self.base_url}/friendrecomendation'
        response = self.session.get(url, params=params).json()
        return response

    def messageIdToURL(self, params):
        url = f'{self.base_url}/msgidtourl'
        response = self.session.get(url, params=params).json()
        return response

    def convertParseUrlToExtension(self, params):
        url = f'{self.base_url}/convertparseurl'
        response = self.session.get(url, params=params).json()
        return response

    def webp2mp4(self, params):
        url = f'{self.base_url}/webp2mp4'
        response = self.session.get(url, params=params).json()
        return response

    def mp42gif(self, params):
        url = f'{self.base_url}/mp42gif'
        response = self.session.get(url, params=params).json()
        return response

    def vid2Apng(self, params):
        url = f'{self.base_url}/video2apng'
        response = self.session.get(url, params=params).json()
        return response

    def smulePost(self, params):
        url = f'{self.base_url}/smuledl'
        response = requests.get(url, params=params).json()
        return response

    def smuleProfile(self, params):
        url = f'{self.base_url}/smuleprofile'
        response = requests.get(url, params=params).json()
        return response

    def instagramPost(self, params):
        url = f'{self.base_url}/instapost'
        response = self.session.get(url, params=params).json()
        return response

    def instagramStory(self, params):
        url = f'{self.base_url}/instastory'
        response = self.session.get(url, params=params).json()
        return response

    def instaStalker(self,params):
        url = f'{self.base_url}/instastalker'
        response = self.session.get(url, params=params).json()
        return response

    def instagramProfileDetails(self,params):
        url = f"{self.base_url}/instaprofile"
        response = self.session.get(url, params=params).json()
        return response

    def downloadXpost(self,params): #Twitter
        url = f'{self.base_url}/twitterdl'
        response = self.session.get(url, params=params).json()
        return response

    def tweetProfile(self, params):
        url = f'{self.base_url}/tweetprofile'
        response = requests.get(url, params=params).json()
        return response

    def downloadFacebook(self,params):
        url = f'{self.base_url}/facebookdl'
        response = self.session.get(url, params=params).json()
        return response

    def downloadTiktok(self,params):
        url = f'{self.base_url}/tiktokdl'
        response = self.session.get(url, params=params).json()
        return response

    def tiktokProfile(self,params):
        url = f'{self.base_url}/tiktok'
        response = self.session.get(url, params=params).json()
        return response

    def combineImages(self,filenamed, files, data):
        url = f'{self.base_url}/combine'
        headers= {"apikey": self.apikey}
        response = requests.post(url, headers=headers, files=files, data={'urls': data['urls']})
        if response.status_code == 200:
            with open(f'{filenamed}', 'wb') as f:
                f.write(response.content)
            return f"Image successfully saved to: {filenamed}"
        else:
            return f"Failed to save image. Status code: {response.status_code}, Response: {response.text}"

    def faceSwap(self, files=None,params=None):
        url = f'{self.base_url}/swapface'
        headers= {"apikey": self.apikey}
        response = requests.post(url, files=files, params=params, headers=headers)
        return response.json()

    def youtubedl(self,params):
        url = f'{self.base_url}/youtubedl'
        response = self.session.get(url, params=params).json()
        return response

    def youtubeQuery(self,params):
        url = f'{self.base_url}/youtubequery'
        response = self.session.get(url, params=params).json()
        return response

    def youtubeTrendingVideo(self,params):
        url = f'{self.base_url}/ytvideotrend'
        response = self.session.get(url, params=params).json()
        return response

    def youtubeTrendingMusic(self,params):
        url = f'{self.base_url}/ytmusictrend'
        response = self.session.get(url, params=params).json()
        return response

    def fancy(self,params):
        url = f'{self.base_url}/fancy'
        response = self.session.get(url, params=params).json()
        return response

    def github(self,params):
        url = f'{self.base_url}/github'
        response = self.session.get(url, params=params).json()
        return response

    def githubClone(self,params):
        url = f'{self.base_url}/githubdl'
        headers = {
            "apikey": self.apikey
        }
        response = requests.post(url, json=params,headers=headers).json()
        return response

File: usage/javascript/exapi.js
const fs = require('fs');
const path = require('path');
const FormData = require('form-data');
const fetch = require('node-fetch');


class ExecrossAPI {
    constructor(base_url, apikey) {
        this.base_url = 'https://execross.com/api/v3';
        this.apikey = 'forexecman';

    }

    async makeRequest(url, params = {}, method = 'GET') {
        try {
            const queryString = new URLSearchParams(params).toString();
            const fullUrl = `${this.base_url}${url}?${queryString}`;
            const response = await fetch(fullUrl, {
                method: method,
                headers: {
                    'apikey': this.apikey,
                    'Content-Type': 'application/json'
                }
            });
            if (!response.ok) {
                const errorData = await response.json().catch(() => ({ message: 'Unknown error' }));
                return {
                    status: response.status,
                    data: null,
                    error: errorData
                };
            }
            const data = await response.json();
            return {
                status: response.status,
                data: data,
                error: null
            };
        } catch (error) {
            return {
                status: 500,
                data: null,
                error: { message: 'Network error', details: error.message }
            };
        }
    }

    async getProxies() {
        const endpoint = '/proxies';
        const result = await this.makeRequest(endpoint);
        if (result.status === 200 && result.data.status === 200) {
            const proxies = result.data.result.proxies;
            return {
                status: 200,
                data: _.sample(proxies).split(':')[0],
                error: null
            };
        }
        return result;
    }

    async getQR(params) {
        const url = '/loginqr';
        const url2 = '/reqpin';
        const url3 = '/reqtoken';
        const response = await this.makeRequest(url, params);
        if (response.status === 200 && response.data.status === 200) {
            const params2 = {
                apikey: this.apikey,
                session: response.data.result.session
            };
            const response2 = await this.makeRequest(url2, params2);
            if (response2.status === 200 && response2.data.result.authToken) {
                return response2;
            } else {
                return await this.makeRequest(url3, params2);
            }
        }
        return response;
    }

    async getQRV1(params) {
        const url = '/loginqrv1';
        const url2 = '/reqpinv1';
        const url3 = '/reqtokenv1';
        const response = await this.makeRequest(url, params);
        if (response.status === 200 && response.data.status === 200) {
            const params2 = {
                apikey: this.apikey,
                session: response.data.result.session
            };
            const response2 = await this.makeRequest(url2, params2);
            if (response2.status === 200 && response2.data.result.authToken) {
                return response2;
            } else {
                return await this.makeRequest(url3, params2);
            }
        }
        return response;
    }

    async getEmail(params) {
        const url = '/loginemail';
        const url2 = '/reqemailtoken';
        const response = await this.makeRequest(url, params);
        if (response.status === 200 && response.data.result.pincode) {
            return await this.makeRequest(url2, { apikey: this.apikey });
        }
        return response;
    }

    async getEmailV1(params) {
        const url = '/loginemailv1';
        const url2 = '/reqemailtokenv1';
        const response = await this.makeRequest(url, params);
        if (response.status === 200 && response.data.result.pincode) {
            return await this.makeRequest(url2, { apikey: this.apikey });
        }
        return response;
    }

    async exampleAddFriendPrimary(params) {
        return await this.makeRequest('/addfriend', params);
    }

    async exampleAddFriendSecondary(params) {
        return await this.makeRequest('/addfriend', params);
    }

    async getStory(params) {
        return await this.makeRequest('/linestory', params);
    }

    async getPost(params) {
        return await this.makeRequest('/linepost', params);
    }

    async friendRecomendation(params) {
        return await this.makeRequest('/friendrecommendation', params);
    }

    async messageIdToURL(params) {
        return await this.makeRequest('/msgidtourl', params);
    }

    async convertParseUrlToExtension(params) {
        return await this.makeRequest('/convertparseurl', params);
    }

    async webp2mp4(params) {
        return await this.makeRequest('/webp2mp4', params);
    }

    async mp42gif(params) {
        return await this.makeRequest('/mp42gif', params);
    }

    async vid2Apng(params) {
        return await this.makeRequest('/video2apng', params);
    }

    async downloadSmule(params) {
        return await this.makeRequest('/smuledl', params);
    }

    async instagramPost(params) {
        return await this.makeRequest('/instapost', params);
    }

    async instaStory(params) {
        return await this.makeRequest('/instastory', params);
    }

    async instaStalker(params) {
        return await this.makeRequest('/instastalker', params);
    }

    async instagramProfileDetails(params) {
        return await this.makeRequest('/instaprofiledetails', params);
    }

    async downloadXpost(params) {
        return await this.makeRequest('/twitterdl', params);
    }

    async downloadFacebook(params) {
        return await this.makeRequest('/facebookdl', params);
    }

    async downloadTiktok(params) {
        return await this.makeRequest('/tiktokdl', params);
    }

    async combineImages(imgPath, filePaths, urls) {
        const url = '/combine';
        const form = new FormData();
        filePaths.forEach(filePath => {
            if (!fs.existsSync(filePath)) {
                console.error(`File not found: ${filePath}`);
                return;
            }
            
            form.append('images', fs.createReadStream(filePath), {
                filename: path.basename(filePath),
                contentType: 'image/jpeg'
            });
        });
        form.append('urls', JSON.stringify(urls));
        try {
            const response = await fetch(`${this.base_url}${url}`, {
                method: 'POST',
                headers: {
                    'apikey': this.apikey,
                    ...form.getHeaders()
                },
                body: form
            });
            const contentType = response.headers.get('content-type');
            if (response.ok) {
                if (contentType.includes('application/json')) {
                    const result = await response.json();
                    return {
                        status: response.status,
                        message: 'Images successfully uploaded',
                        data: result
                    };
                } else if (contentType.includes('image/')) {
                    const buffer = await response.buffer();
                    fs.writeFileSync(`${imgPath}`, buffer); 
                    return {
                        status: response.status,
                        message: `Image successfully saved at: ${imgPath}`
                    };
                } else {
                    const text = await response.text();
                    return {
                        status: response.status,
                        message: 'Received unexpected content type',
                        data: text
                    };
                }
            } else {
                const errorText = await response.text();
                return {
                    status: response.status,
                    message: `Failed to upload images. Status code: ${response.status}`,
                    error: errorText
                };
            }
        } catch (error) {
            return {
                status: 500,
                message: 'Error occurred during the request',
                error: error.message
            };
        }
    }

    async swapFaceWithURLs(params) {
        const apiParams = {
            originImageURL: params.originImageURL, 
            faceImageURL: params.faceImageURL
        };
        return await this.makeRequest('/swapface', apiParams, 'POST');
    }
    
    async swapFaceWithFiles(originImagePath, faceImagePath) {
        const formData = new FormData();
        formData.append('originImage', fs.readFileSync(originImagePath), 'original_image.jpg');
        formData.append('faceImage', fs.readFileSync(faceImagePath), 'face_image.jpg');

        try {
            const fullUrl = `${this.base_url}/swapface`;
            const response = await fetch(fullUrl, {
                method: 'POST',
                headers: {
                    'apikey': this.apikey
                },
                body: formData
            });

            if (!response.ok) {
                const errorData = await response.json().catch(() => ({ message: 'Unknown error' }));
                return {
                    status: response.status,
                    data: null,
                    error: errorData
                };
            }

            const data = await response.json();
            return {
                status: response.status,
                data: data,
                error: null
            };
        } catch (error) {
            return {
                status: 500,
                data: null,
                error: { message: 'Network error', details: error.message }
            };
        }
    }
}

module.exports = ExecrossAPI;
